frontend/apps/reader/modules/readertoc.lua

```diff
@@ -280,6 +389,9 @@ function ReaderToc:onShowToc()
         for _,v in ipairs(self.toc) do
             v.text = self.toc_indent:rep(v.depth-1)..self:cleanUpTocTitle(v.title)
             v.mandatory = v.page
+            if v.orig_page then -- bogus page fixed: show original page number
+                v.mandatory = T("(%1) %2", v.orig_page, v.page)
```

Comment:
Bidi?

Reply:
Looks like BiDi will handle it automagically:
<kbd>![image](https://user-images.githubusercontent.com/24273478/75571427-79c89880-5a59-11ea-8e18-261cb103c635.png)</kbd>
(wider screen width, so not same numbers as first post screenshots)